client/src/main/java/org/spine3/time/Formats.java

```diff
@@ -0,0 +1,211 @@
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.time;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+import static org.spine3.time.Time.NANOS_PER_MICROSECOND;
+import static org.spine3.time.Time.NANOS_PER_MILLISECOND;
+
+/**
+ * Utilities and constants for working with date/time formats.
+ *
+ * @author Alexander Yevsyukov
+ */
+final class Formats {
+
+    static final String DATE_FORMAT = "yyyy-MM-dd";
+    static final String DATE_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";
+    static final String HOURS_AND_MINUTES_FORMAT = "%02d:%02d";
+    static final String TIME_FORMAT = "HH:mm:ss";
+
+    static final char TIME_SEPARATOR = 'T';
+    static final char SUB_SECOND_SEPARATOR = '.';
+    static final char TIME_VALUE_SEPARATOR = ':';
+    static final char PLUS = '+';
+    static final char MINUS = '-';
+    static final char UTC_ZONE_SIGN = 'Z';
+
+    private static final ThreadLocal<DateFormat> dateTimeFormat =
+            new ThreadLocal<DateFormat>() {
+                @Override
+                protected DateFormat initialValue() {
+                    return createDateTimeFormat(TimeZone.getDefault());
+                }
+            };
+
+    private static final ThreadLocal<DateFormat> timeFormat =
+            new ThreadLocal<DateFormat>() {
+                @Override
+                protected DateFormat initialValue() {
+                    return createTimeFormat(TimeZone.getDefault());
+                }
+            };
+
+    private static final ThreadLocal<DateFormat> dateFormat =
+            new ThreadLocal<DateFormat>() {
+                @Override
+                protected DateFormat initialValue() {
+                    return createDateFormat(TimeZone.getDefault());
+                }
+            };
+
+    /**
+     * Names of arguments in preconditions checks.
+     */
+    enum Parameter {
+        hours,
+        minutes,
+        seconds,
+        millis,
+        nanos
+    }
+
+    private Formats() {
+        // Prevent instantiation of this utility class.
+    }
+
+    /**
+     * Creates format for local date.
+     */
+    private static DateFormat createDateFormat(TimeZone timeZone) {
+        final SimpleDateFormat sdf = createDigitOnlyFormat(DATE_FORMAT);
```

Comment:
We cannot reuse `DateFormat`s between threads. That's why they are accessed via `ThreadLocal`s.